src/data-workspace/display/display.js

```diff
@@ -3,30 +3,60 @@ import i18n from '@dhis2/d2-i18n'
 import { NoticeBox, CircularLoader } from '@dhis2/ui'
 import PropTypes from 'prop-types'
 import React, { useEffect } from 'react'
+import { useSelectionContext } from '../../selection-context/index.js'
+import {
+    compareFixedPeriodLength,
+    getFixedPeriodsForTypeAndDateRange,
+    PERIOD_SHORTER,
+} from '../../shared/index.js'
 import styles from './display.module.css'
 import { Table } from './table.js'
 
 const query = {
     dataSetReport: {
         resource: 'dataSetReport',
-        params: ({ ds, pe, ou }) => ({
-            ds,
-            pe,
-            ou,
-        }),
+        params: ({ dataSetId, period, orgUnit, workflow }) => {
+            let periodIds = [period.id]
+            const { dataSets } = workflow
+            const selectedDataSet = dataSets.find(({ id }) => id === dataSetId)
+            const { periodType: dataSetPeriodType } = selectedDataSet
+
+            const isDataSetPeriodShorter =
+                compareFixedPeriodLength(
+                    workflow.periodType,
+                    dataSetPeriodType
+                ) === PERIOD_SHORTER
+
+            if (isDataSetPeriodShorter) {
+                periodIds = getFixedPeriodsForTypeAndDateRange(
+                    dataSetPeriodType,
+                    period.startDate,
+                    period.endDate
+                )
+            }
+
+            return {
+                ds: dataSetId,
+                pe: periodIds.join(','),
```

Comment:
`.join` shouldn't be needed here as the runtime does it for us [here](https://github.com/dhis2/app-runtime/blob/8dd5885dec1654e39e87aa33c6ff431ccdae6876/services/data/src/links/RestAPILink/queryToResourcePath.ts#L10-L12)